src/test/java/ParserTest.java

```diff
@@ -0,0 +1,127 @@
+
+        String result = "Got it. I've added this task:\n" +
+                "[T][ ] read book";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+    @Test
+    void testDeadlineDash() throws DukeException {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+
+        parser.createTask("deadline return book /by 2023-12-11");
+
+        String result = "Got it. I've added this task:\n" +
+                "[D][ ] return book  (by: Dec 11 2023)";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+    @Test
+    void testDeadlineSlash() throws DukeException {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+
+        parser.createTask("deadline return book /by 1/12/2022 1800");
+
+        String result = "Got it. I've added this task:\n" +
+                "[D][ ] return book  (by: Dec 01 2022)";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+    @Test
+    void testDeadlineOthers() throws DukeException {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+
+        parser.createTask("deadline return book /by December 2022");
+
+        String result = "Got it. I've added this task:\n" +
+                "[D][ ] return book  (by:  December 2022)";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+    @Test
+    void testEventDash() throws DukeException {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+
+        parser.createTask("event return book /from 2022-12-11 /to 2022-12-12");
+
+        String result = "Got it. I've added this task:\n" +
+                "[E][ ] return book  (from: Dec 11 2022 to: Dec 12 2022)";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+    @Test
+    void testEventSlash() throws DukeException {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+
+        parser.createTask("event return book /from 11/12/2022 1800 /to 12/12/2022 1800");
+
+        String result = "Got it. I've added this task:\n" +
+                "[E][ ] return book  (from: Dec 11 2022 to: Dec 12 2022)";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+    @Test
+    void testEventOthers() throws DukeException {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+
+        parser.createTask("event return book /from Tuesday /to Thursday");
+
+        String result = "Got it. I've added this task:\n" +
+                "[E][ ] return book  (from: Tuesday to: Thursday)";
+
+        assertEquals(result, outputStreamCaptor.toString()
+                .trim());
+
+    }
+
+
+}
```

Comment:
Good effort on implementing the test cases to test different date formats. Perhaps the test name can include something about date format (rather than dash, slash and others) to make it clear that test focuses on testing date formats.